### ISSUE TITLE
Confusing record type error

### ISSUE THREAD

**Issue by agrafix** [elm-lang/error-message-catalog#30]
I just came across this error:

```
-- TYPE MISMATCH -------------------------------- ./elmsrc/Comp/NewCrossover.elm

The 1st argument to function `inputInt'` has an unexpected type.

82|           inputInt'
83|>            { class = "form-control"
84|>            , name = "rail-dist"
85|>            , placeholder = Just "Spurweite in mm"
86|>            , value = m.railDist
87|>            , min = 0
88|>            , max = 2000
89|>            , step = 5
90|>            , update =
91|>                { fieldUpdate
92|>                    | onInput <-
93|>                        Just (\val ->
94|>                            Just (Signal.message addr (SetRailDist val))
95|>                        )
96|>                }
97|>            }

As I infer the type of values flowing through your program, I see a conflict
between these two types:

    Maybe Int

    number
```

The type error does not make it clear what went wrong. For example I set `min = 0`, but the type of min is `Maybe Int`. The expected error should contain something like `Line 87, field min`.

**Comment by rtfeldman** [comment]
Seems related to https://github.com/evancz/error-message-catalog/issues/23

**Comment by evancz** [comment]
I think this will be resolved with 0.16 but I have no way to know without a http://sscce.org/ to test. I'm going to close under the suspicion that it is resolved in a nice way, and in the future, please make it so I can test!
